components/script/script_thread.rs

```diff
@@ -2081,7 +2082,15 @@ impl ScriptThread {
             referrer_policy: load_data.referrer_policy,
             referrer_url: load_data.referrer_url,
             source: RequestSource::Window(self.custom_message_chan.clone())
-        }, LoadConsumer::Listener(response_target), None)).unwrap();
+        };
+
+        if load_data.url.scheme() == "blob" {
+            let msg = FileManagerThreadMsg::LoadBlob(net_load_data, LoadConsumer::Listener(response_target));
+            self.resource_threads.send(msg).unwrap();
+        } else {
+            let msg = CoreResourceMsg::Load(net_load_data, LoadConsumer::Listener(response_target), None);
```

Comment:
Yep, that is one way. But I remembered that we discussed this before and thought `CoreResource` thread handler might be too busy thus clogged?